roc/log_test.go

```diff
@@ -0,0 +1,100 @@
+package roc
+
+import (
+	"io/ioutil"
+	"log"
+	"testing"
+	"time"
+)
+
+const defaultLogLevel LogLevel = LogError
+
+type testWriter struct {
+	ch chan string
+}
+
+func makeTestWriter() testWriter {
+	return testWriter{
+		ch: make(chan string, 1),
+	}
+}
+
+func (tw testWriter) Write(buf []byte) (int, error) {
+	select {
+	case tw.ch <- string(buf):
+	default:
+	}
+	return len(buf), nil
+}
+
+func (tw testWriter) wait() string {
+	select {
+	case s := <-tw.ch:
+		return s
+	case <-time.After(time.Minute):
+		return ""
+	}
+}
+
+func Test_roc_log_default(t *testing.T) {
+	setupFuncs := []func(){
+		func() {},
+		func() { SetLoggerFunc(nil) },
+		func() { SetLogger(nil) },
+	}
+
+	SetLogLevel(LogDebug)
+	defer SetLogLevel(defaultLogLevel)
+
+	for n, setupFn := range setupFuncs {
+		tw := makeTestWriter()
+
+		log.SetOutput(&tw)
+		defer log.SetOutput(ioutil.Discard)
+
+		setupFn()
+
+		ctx, _ := OpenContext(&ContextConfig{})
+		ctx.Close()
+
+		if tw.wait() == "" {
+			t.Fatalf("test %v failed", n)
```

Comment:
maybe it makes sense to say why exactly it failed: "Error: expected logs, didn't get them before timeout" or soemthing more articulate

Reply:
Agree, added message here and in other tests.